ghmirror/data_structures/requests_cache.py

```diff
@@ -0,0 +1,64 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2020
+# Author: Maha Ashour <mashour@redhat.com>
+
+"""
+Implements caching backend
+"""
+
+import os
+from abc import abstractmethod
+
+from ghmirror.data_structures.monostate import InMemoryCache
+from ghmirror.data_structures.redis_data_structures import RedisCache
+
+
+class RequestsCache:
+    """
+    Inherits either InMemoryCache or Redis Cache object
```

Comment:
Inheritance is not what is going on here.

Reply:
ah i forgot that doc string. would it be correct to call this an abstract class that is implemented by both InMemoryCache or Redis Cache?

Reply:
No. None of them inherit from this one.

Reply:
ok, i changed the doc string to just say it instantiate one of two objects